create_admin_user.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+# Usage create_admin_user.sh -u user -p password -h host -k private_key
+#
+# Description of required command line arguments:
+#
+#   -u: admin user name
+#   -p: admin user password
+#   -h: host name or IP address
+#   -k: path to private SSH key file
+#
+# Usage example: create_admin_user.sh -u xrd -p secret -h host -k private_key_file
+
+usage() {
+  echo "Usage: create_admin_user.sh -u xrd -p secret -h host -k private_key_file"
+}
+
+exit_abnormal() {
+  usage
+  exit 1
+}
+
+create_user() {
+    user=$1
+    pass=$2
+    host=$3
+    ssh_key=$4
+    os_name=$(ssh -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "$ssh_key" root@"$host" "grep '^NAME' /etc/os-release")
+
+    # create user and add to sudo/root group
+    if echo "$os_name" | grep 'Ubuntu'; then
+      ssh -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "$ssh_key" root@"$host" "useradd $user --group sudo -p $pass"
+    fi
+    if echo "$os_name" | grep 'Centos'; then
+      ssh -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "$ssh_key" root@"$host" "useradd --system --home /var/lib/xroad --no-create-home --shell /bin/bash --user-group --comment \"X-Road system user\" xroad"
+      ssh -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "$ssh_key" root@"$host" "mkdir /etc/xroad"
+      ssh -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "$ssh_key" root@"$host" "chown xroad:xroad /etc/xroad"
+      ssh -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "$ssh_key" root@"$host" "chmod 751 /etc/xroad"
+      ssh -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "$ssh_key" root@"$host" "touch /etc/xroad/db.properties"
+      ssh -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "$ssh_key" root@"$host" "chown xroad:xroad /etc/xroad/db.properties"
+      ssh -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "$ssh_key" root@"$host" "chmod 640 /etc/xroad/db.properties"
```

Comment:
Lines 35-40: where do all these commands come from, are they from the security server installation script or somewhere else? In any case they seem to attain for the case where no security server package is yet installed at all. But then, the later addition to xroad security server specific groups (line 50) would also be bound to fail, simply because the groups do not exist yet?

This seems to me it is either doing too much with these xroad specific file hierarchies, or too little -- when the security server software indeed has not yet been installed. How would the installer behave when finding these files and folders present, is it able to continue with the correct installation?

Reply:
All non-existing groups are now created and this is taken from chapter 2.5.1 in: https://github.com/nordic-institute/X-Road/blob/develop/doc/Manuals/ig-ss_x-road_v6_security_server_installation_guide_for_rhel.md

Reply:
![wordsfail](https://user-images.githubusercontent.com/378754/113777931-6a278300-9734-11eb-91d9-bbd25009cbad.png)

^^ 
I still have the same impression -- it is either doing too much or too little. Please explain what it is that is actually being done here, and most importantly, why that is necessary for the user story, what makes _optional_ step 2.5.1 titled _customizing the database properties_ from cited manual even needed here?

Reply:
For RHEL, the normal way to add the admin user is after installing the packages described in: https://github.com/nordic-institute/X-Road/blob/develop/doc/Manuals/ig-ss_x-road_v6_security_server_installation_guide_for_rhel.md#28-package-installation

The call described in "2.5.1 Customize the Database Properties" is used if the user needs to be created before the security-server packages have been installed (because the script xroad-add-admin-user comes with the installation) and installation of these packages is out of the scope of this user story